core/lib/bin_metadata/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+use serde::Serialize;
+use vise::{EncodeLabelSet, Info, Metrics};
+
+use self::values::BIN_METADATA;
+
+pub mod values {
+    use super::BinMetadata;
+    include!(concat!(env!("OUT_DIR"), "/metadata_values.rs"));
+}
+
+/// Metadata of the compiled binary.
+#[derive(Debug, EncodeLabelSet, Serialize)]
+pub struct BinMetadata {
+    pub rustc_version: &'static str,
+    pub rustc_commit_hash: Option<&'static str>,
+    pub rustc_commit_date: Option<&'static str>,
+    pub rustc_channel: &'static str,
+    pub host: &'static str,
+    pub llvm: Option<&'static str>,
+    pub git_branch: Option<&'static str>,
+    pub git_revision: Option<&'static str>,
+}
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "rust")]
```

Comment:
Eh, new `git_*` metrics probably shouldn't be grouped into the `rust_info` INFO metric. Optimally, I'd add a new INFO metric with these two labels and name it like `git_info` or `version_control_info`.

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3193/commits/b6bdda43ed6164bccc3025e95f5f95c469501306